src/usdb_syncer/gui/mw.py

```diff
@@ -149,8 +150,18 @@ def _setup_toolbar(self) -> None:
     def _setup_shortcuts(self) -> None:
         gui_utils.set_shortcut("Ctrl+.", self, lambda: DebugConsole(self).show())
 
-    def _setup_song_dir(self) -> None:
-        self.lineEdit_song_dir.setText(str(settings.get_song_dir()))
+    def _setup_song_dir(self, songpath: str | None) -> None:
+        self.song_dir = settings.get_song_dir()
```

Comment:
song_dir is not kept in sync with lineEdit_song_dir. And more importantly: --song-path is not used for downloading.
It would be much easier if there was a single source of truth. What if --songpath also updates the setting? I understand that's less versatile, but also much less error-prone.

Reply:
Not sure if that's not confusing. I would expect that passing `--songpath`*temporarily` overwrites the song_dir, and if I restart the syncer with argument, it will start with the song_dir stored in the settings.

Reply:
I don't really mind either way, as long as it's consistent and the code is clear.